umbraco/Umbraco-CMS
# Make delivery API indexers and handlers public (sealed) so they can be overridden by custom ones

### Prerequisites


### Description

All delivery API indexers, selectors, filters and sorts must be public; otherwise they cannot be overridden by custom implementations.

### Testing this PR

1. Add the code snippet below (contains a custom children selector and a composer).
2. Use the debugger to verify that a "children" fetch is handled by the custom children selector.
3. Comment out the composer and rebuild.
4. Use the debugger to verify that a "children" fetch is no longer handled by the custom children selector.

```cs
using Umbraco.Cms.Api.Delivery.Querying;
using Umbraco.Cms.Api.Delivery.Querying.Selectors;
using Umbraco.Cms.Core.Composing;
using Umbraco.Cms.Core.DeliveryApi;
using Umbraco.Cms.Core.PublishedCache;

namespace Umbraco.Cms.Web.UI.Custom;

public class MyChildrenSelector : QueryOptionBase, ISelectorHandler
{
    private const string ChildrenSpecifier = "children:";

    public MyChildrenSelector(IPublishedSnapshotAccessor publishedSnapshotAccessor, IRequestRoutingService requestRoutingService)
        : base(publishedSnapshotAccessor, requestRoutingService)
    {
    }

    /// <inheritdoc />
    public bool CanHandle(string query)
        => query.StartsWith(ChildrenSpecifier, StringComparison.OrdinalIgnoreCase);

    /// <inheritdoc/>
    public SelectorOption BuildSelectorOption(string selector)
    {
        var fieldValue = selector.Substring(ChildrenSpecifier.Length);
        Guid? id = GetGuidFromQuery(fieldValue);

        return new SelectorOption
        {
            FieldName = "parentId",
            Value = id.ToString() ?? string.Empty
        };
    }
}

public class MyComposer : IComposer
{
    public void Compose(IUmbracoBuilder builder)
    {
        builder.SelectorHandlers().Exclude<ChildrenSelector>();
    }
}
```
